pkg/syncer/syncertest/fake/client.go

```diff
@@ -214,3 +219,8 @@ func (c *Client) RESTMapper() meta.RESTMapper {
 func (c *Client) Storage() *MemoryStorage {
 	return c.storage
 }
+
+// SubResource implements client.Client
+func (c *Client) SubResource(_ string) client.SubResourceClient {
+	panic("SubResource not implemented for FakeClient")
```

Comment:
this should be trivial to impl. it's the same as Status() just without a hard coded subresource name.

Reply:
Was hoping I could avoid fully refactoring all of these interfaces since these aren't used currently, but I guess I'll take a stab at it. Probably not difficult to do but properly refactoring this will be a larger change.

Reply:
Refactored